src/test/java/io/confluent/connect/jdbc/source/integration/MSSQLDateTimeIT.java

```diff
@@ -0,0 +1,314 @@
+import static io.confluent.connect.utils.licensing.LicenseConfigUtil.CONFLUENT_TOPIC_REPLICATION_FACTOR_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.CONNECTOR_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.KEY_CONVERTER_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.TASKS_MAX_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.VALUE_CONVERTER_CLASS_CONFIG;
+
+@Category(IntegrationTest.class)
+public class MSSQLDateTimeIT extends BaseConnectorIT {
+
+    private static final Logger log = LoggerFactory.getLogger(MSSQLDateTimeIT.class);
+    private static final String CONNECTOR_NAME = "JdbcSourceConnector";
+    private static final int NUM_RECORDS_PRODUCED = 1;
+    private static final long CONSUME_MAX_DURATION_MS = TimeUnit.MINUTES.toMillis(2);
+    private static final int TASKS_MAX = 3;
+    private static final String MSSQL_URL = "jdbc:sqlserver://0.0.0.0:1433";
+    private static final String MSSQL_Table = "TestTable";
+    private static final String TOPIC_PREFIX = "test-";
+    private static final List<String> KAFKA_TOPICS = Collections.singletonList(TOPIC_PREFIX + MSSQL_Table );
+
+    private Map<String, String> props;
+
+    private static final String USER = "sa"; // test creds
+    private static final String PASS = "reallyStrongPwd123"; // test creds
+
+    private Connection connection;
+
+    @ClassRule
+    public static final FixedHostPortGenericContainer mssqlServer =
+            new FixedHostPortGenericContainer<>("microsoft/mssql-server-linux:latest")
+                .withEnv("ACCEPT_EULA","Y")
+                .withEnv("SA_PASSWORD","reallyStrongPwd123")
+                .withFixedExposedPort(1433, 1433);
+
+    @Before
+    public void setup() throws Exception {
+        //Create table
+        Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
+        connection = DriverManager.getConnection(MSSQL_URL, USER, PASS);
+        startConnect();
+
+    }
+
+    @After
+    public void close() throws SQLException {
+        connect.deleteConnector(CONNECTOR_NAME);
+        connection.close();
+        // Stop all Connect, Kafka and Zk threads.
+        stopConnect();
+    }
+
+    /**
+     * Verify that after a number of poll intervals the number of records in topic
+     * still equal to the number of records written to the MSSQL Server instance.
+     *
+     * This test is for a datetime to datetime2 comparison error in MSSQL Server (2016 and after).
+     * Datetime is converted recursively to a higher precision (eg- 3.33 MS would be 3.33333 ),
+     * while java.sql.Timestamp converts by tacking on 0s (3.33000).
+     */
+    // Verify connect error thrown
+    @Test
+    public void verifyDateTimeTSModeKillsTasks() throws Exception {
+        // Setup up props for the source connector
+        props = configProperties(true);
+        createDateTimeTable("DATETIME", "VARCHAR(255)", true);
```

Comment:
This is goofy to read, can you make the method sig/params more intuitive to read?